lib/audio_service_web.dart

```diff
@@ -0,0 +1,260 @@
+      case 'connect':
+        // No-op not really a reason for connect in web
+        break;
+      case 'disconnect':
+        // No-op not really a reason for disconnect in web
+        break;
+      case 'isRunning':
+        return started;
+      case 'rewind':
+        return backgroundChannel.invokeMethod('onRewind');
+      case 'fastForward':
+        return backgroundChannel.invokeMethod('onFastForward');
+      case 'skipToPrevious':
+        return backgroundChannel.invokeMethod('onSkipToPrevious');
+      case 'skipToNext':
+        return backgroundChannel.invokeMethod('onSkipToNext');
+      case 'play':
+        return backgroundChannel.invokeMethod('onPlay');
+      case 'pause':
+        return backgroundChannel.invokeMethod('onPause');
+      case 'stop':
+        return backgroundChannel.invokeMethod('onStop');
+      case 'seekTo':
+        return backgroundChannel.invokeMethod('onSeekTo', [call.arguments]);
+      case 'prepareFromMediaId':
+        return backgroundChannel
+            .invokeMethod('onPrepareFromMediaId', [call.arguments]);
+      case 'playFromMediaId':
+        return backgroundChannel
+            .invokeMethod('onPlayFromMediaId', [call.arguments]);
+      case 'setBrowseMediaParent':
+        return backgroundChannel
+            .invokeMethod('onLoadChildren', [call.arguments]);
+      case 'onClick':
+        //no-op
+        break;
+      case 'addQueueItem':
+        return backgroundChannel
+            .invokeMethod('onAddQueueItem', [call.arguments]);
+      case 'addQueueItemAt':
+        return backgroundChannel.invokeMethod('onQueueItemAt', call.arguments);
+      case 'removeQueueItem':
+        return backgroundChannel
+            .invokeMethod('onRemoveQueueItem', [call.arguments]);
+      case 'updateQueue':
+        return backgroundChannel
+            .invokeMethod('onUpdateQueue', [call.arguments]);
+      case 'updateMediaItem':
+        return backgroundChannel
+            .invokeMethod('onUpdateMediaItem', [call.arguments]);
+      case 'prepare':
+        return backgroundChannel.invokeMethod('onPrepare');
+      case 'playMediaItem':
+        return backgroundChannel
+            .invokeMethod('onPlayMediaItem', [call.arguments]);
+      case 'skipToQueueItem':
+        return backgroundChannel
+            .invokeMethod('onSkipToMediaItem', [call.arguments]);
+      case 'setRepeatMode':
+        return backgroundChannel
+            .invokeMethod('onSetRepeatMode', [call.arguments]);
+      case 'setShuffleMode':
+        return backgroundChannel
+            .invokeMethod('onSetShuffleMode', [call.arguments]);
+      case 'setRating':
+        return backgroundChannel.invokeMethod('onSetRating',
+            [call.arguments['rating'], call.arguments['extras']]);
+      case 'setSpeed':
+        return backgroundChannel.invokeMethod('onSetSpeed', [call.arguments]);
+      // TODO: CustomActions can't work through the IsolateNameServer we need some other mechanism
```

Comment:
Actually, custom actions are currently implemented just like pass throughs just like the above pass throughs.

On Android, it is handled by this default case:

```java
    backgroundHandler.channel.invokeMethod(call.method, call.arguments, result);
```

And on iOS it is handled by this default case:

```objc
    [backgroundChannel invokeMethod:call.method arguments:call.arguments result: result];
```

Reply:
Ok awesome! I can forward that through. I was mostly worried about the `customEventStream` because that currently works off of the name server. But I just quickly added it in and everything seems to work. I now realize that `customActions` are not the same thing as `customEvents`. I was able to replace the `SendPort` call with a direct add to the behavior subject, so that's working pretty well.